src/debugger/debugAdapter.ts

```diff
@@ -24,21 +24,19 @@ import { SwiftToolchain } from "../toolchain/toolchain";
  * Class managing which debug adapter we are using. Will only setup lldb-vscode/lldb-dap if it is available.
  */
 export class DebugAdapter {
-    private static debugAdapaterExists = false;
-
     /** Debug adapter name */
-    public static get adapterName(): string {
-        return configuration.debugger.useDebugAdapterFromToolchain && this.debugAdapaterExists
+    public static getAdapterName(swiftVersion: Version): "swift-lldb" | "lldb" {
+        return DebugAdapter.getDebugAdapterType(swiftVersion) === "lldb-dap"
             ? "swift-lldb"
             : "lldb";
     }
 
     /** Return debug adapter for toolchain */
-    public static getDebugAdapterType(swiftVersion: Version): "lldb-vscode" | "lldb-dap" {
+    public static getDebugAdapterType(swiftVersion: Version): "lldb" | "lldb-dap" {
         return swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0)) &&
             configuration.debugger.useDebugAdapterFromToolchain
             ? "lldb-dap"
-            : "lldb-vscode";
+            : "lldb";
```

Comment:
so it seems like this string needs to match the launch configuration type provided by CodeLLDB. We should make that more clear with some comments, and probably worth introducing some CODELLDB_TYPE constant to make intent clear